metrics/ci_integrations/lib/client/github_actions/github_actions_client.dart

```diff
@@ -468,6 +472,131 @@ class GithubActionsClient with LoggerMixin {
     }
   }
 
+  /// Fetches a [GithubToken] by the given [auth].
+  ///
+  /// Throws an [ArgumentError] if the given [auth] is `null`.
+  Future<InteractionResult<GithubToken>> fetchToken(
+    AuthorizationBase auth,
+  ) {
+    ArgumentError.checkNotNull(auth, 'auth');
+
+    final url = UrlUtils.buildUrl(
+      githubApiUrl,
+      path: 'user',
+    );
+
+    final requestHeaders = {
+      ...headers,
+      ...auth.toMap(),
+    };
+
+    return _handleResponse(
+      _client.get(url, headers: requestHeaders),
+      (_, headers) {
+        final scopes = headers['x-oauth-scopes'] ?? '';
+        final scopesList = scopes.split(',').map((e) => e.trim()).toList();
+
+        final githubToken = GithubToken.fromJson({'scopes': scopesList});
+
+        return InteractionResult.success(result: githubToken);
+      },
+    );
+  }
+
+  /// Fetches a [GithubUser] by the given [name].
+  ///
+  /// Throws an [ArgumentError] if the given [name] is `null`.
+  Future<InteractionResult<GithubUser>> fetchGithubUser(
+    String name,
+  ) {
+    ArgumentError.checkNotNull(name, 'name');
+
+    final url = UrlUtils.buildUrl(
+      githubApiUrl,
+      path: 'users/$name',
+    );
+
+    return _handleResponse(
+      _client.get(url, headers: headers),
+      (json, _) {
+        final githubUser = GithubUser.fromJson(json);
+
+        return InteractionResult.success(result: githubUser);
+      },
+    );
+  }
+
+  /// Fetches a [GithubRepository]
+  /// by the given [auth], [repositoryName] and [repositoryOwnerName].
+  ///
+  /// Throws an [ArgumentError] if at least one of the given parameters
+  /// is `null`.
+  Future<InteractionResult<GithubRepository>> fetchGithubRepository({
+    AuthorizationBase auth,
```

Comment:
I believe we can use `auth` provided to the `GithubActionsClient` constructor here. The same suggestion for the `fetchWorkflow` method.